src/snowflake/snowpark/dataframe.py

```diff
@@ -26,6 +26,8 @@
     overload,
 )
 
+import pandas as pd
```

Comment:
this will cause error when pandas is not installed -- pandas is not a required dependency to snowpark

you should use the pandas from `snowflake.connector.options` which is already imported